src/DotVVM.Framework/Resources/Scripts/DotVVM.ts

```diff
@@ -31,6 +31,12 @@ interface IDotvvmViewModelInfo {
 interface IDotvvmViewModels {
     [name: string]: IDotvvmViewModelInfo
 }
+interface IDotvvmStaticCommandResponse {
+    action: string;
+    url: string;
+    result: any;
```

Comment:
The types are wrong, as all three properties are optional.

Reply:
@exyi What are the correct ones? If they are optional, do you think they should not be declared?

Reply:
They should be declared as optional: `action?: string`. Or better, it seems that only these cases are valid `{ result: any }`, `{ action: "redirect"; url: string }`, so we can use:

```typescript
type DotvvmStaticCommandResponse =
      { result: any }
    | { action: "redirect"; url: string }
```

There also isn't (AFAIK) a reson to use `interface`, and even if there would, interfaces in typescript are usually not prefixed with `I`.

Reply:
I defined the type as 
```typescript 
type DotvvmStaticCommandResponse = { result?: any; action?: "redirect"; url?: string };
```
The problem is once you access the result property the compiler does not let you access the action property.

Reply:
I have redefined that as a union.